### PR TITLE
chore: bump the mthds floor to 0.11.0 and adapt to the input-form item split

### PR DESCRIPTION
**Pull request by lchoquel**
## What

Moves the `mthds` floor from `>=0.9.0` to `>=0.11.0` and adapts the engine to the input-form descriptor's named/nameless split that `mthds` 0.10.0 introduced.

`InputFormField` stays the named union, but its per-kind models now require `name: str`; the new `InputFormItem` union is their nameless counterpart (`TextItem`, `ObjectItem`, …), which is what a list's `item` holds and the only shape that position accepts. Two re-exported bases were renamed with it — `InputFormFieldBase` → `InputFormItemBase`, `TextValuedFieldBase` → `TextValuedItemBase` — so this is **breaking for pipelex's own importers**, which is why it reaches the changelog as its own entry rather than hiding inside the floor line.

**The wire is unchanged.** This engine already stopped emitting `name` on an item in `bdd853c41`, and the standard recaptured its parity fixtures from that commit. What moved is that the rule is now the shape rather than a convention.

## The one change that is not a rename

`_as_list_item` returned `node.model_copy(update={"name": None})`. `model_copy` does not validate, so under the new shapes that leaves a required `str` slot holding `None`. It now rebuilds the node one layer down through a `TypeAdapter(InputFormItem)`, letting the union's own `kind` discriminator pick the model — going through the union rather than a kind-to-model table keeps the standard's kind list in one place.

That needed a test, because every cheaper guard is blind to it:

- the wire is identical either way — the `item` slot is declared as the nameless model, so the serializer never writes a `name` whatever the object actually is;
- pyright accepts a `*Field` in `item` position too, since each `*Field` subclasses its `*Item`.

So `TestDerivedListItemsAreNameless` asks the derived item what it *is*. Reverting `_as_list_item` to the old line turns both red with `ObjectField(… name=None)` in the failure message — the exact silent wrongness the split exists to remove.

## Fallout from `name` becoming required

- `pipelex/pipeline/vacuous_presence_warnings.py` — a `slot.name is None` guard the split made unreachable, deleted.
- `tests/helpers/input_form.py` and `tests/integration/pipelex/pipeline/test_input_form.py` — a dead `if field.name is not None` filter in two lookups.
- `tests/unit/pipelex/pipeline/test_input_form_models.py` — the descriptor fixture builds its list item as `ObjectItem`, which is now the only thing that position takes.

## Not part of the bump

The test badge was already stale on `dev` (`10952` against an actual `10981`, last written at the v0.53.0 release), so the gap came from commits merged since and would have ridden into this PR as a red CI-only gate. Refreshed here to `10983`, the two new tests included.

## Checks

`make agent-check` green, `make agent-test` green, `.test_durations` refreshed (+2, exactly the new tests), `check-test-badge` green.

## Downstream

`MTHDS_STANDARD_VERSION` did **not** move (still `1.0.0`), so no crate stamp, manifest-constraint warning or fingerprint changes.

`pipelex-server/temporal` touches only `PipeInputFormDescriptor` and `build_input_form`, so it rides the ordinary pipelex pin cascade with no source adaptation.

`pipelex-sdk-python` is a live break, filed as `L-260827-a56c9c`: its floor is `>=0.9.0`, so any fresh resolve there already installs 0.11.0, and `tests/unit/test_validation_contract.py:406` asserts `isinstance(list_node.item, DocumentField)` on what now parses as `DocumentItem`.

Closes L-260827-985b60

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_016nQTe3kpFNsSEGAZHca4fe

---
## Summary by cubic
Moves the `mthds` floor to `>=0.11.0` and adapts to the input-form descriptor's named/nameless split that `mthds` 0.10.0 introduced; the wire output is unchanged. Adds the `bump-mthds` skill documenting the bump procedure and the `upstream_notes.py` helper it uses.

**Breaking for importers**
- `InputFormField` now requires `name: str`, and a list's `item` holds the new nameless `InputFormItem` union with its `TextItem`/`ObjectItem` models.
- Re-exported bases renamed: `InputFormFieldBase` → `InputFormItemBase`, `TextValuedFieldBase` → `TextValuedItemBase`.
- `_as_list_item` rebuilds items through the union's `kind` discriminator, since `model_copy(update={"name": None})` skips validation and leaves a required `str` holding `None`.

**Fallout**
- Scoped the agent `TypeAdapter` rule: the flat ban is now "don't design shapes needing one," with external-contract unions as the sanctioned use.
- Removed dead `name is None` guards in `pipelex/pipeline/vacuous_presence_warnings.py` and the test helpers.
- Added tests pinning derived items to the nameless models, which both the wire output and static typing are blind to.
- `pipelex-sdk-python` breaks downstream (filed as L-260827-a56c9c), while `pipelex-server/temporal` rides the pin cascade unchanged.
- Refreshed the test badge, which was already stale on `dev`.
- Closes L-260827-985b60.

<sup>Written for commit 0b91717de9a677e7d8fa15efbcda100baa5ff871. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Pipelex/pipelex/pull/1156?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by lchoquel** [comment]
@cubic

File: pipelex/pipeline/input_form.py [hunk 1]
```diff
@@ -481,14 +507,25 @@ def _list_item(self, *, name: str, field: ConceptStructureBlueprint, seen: froze
                 return _scalar_field(field_type=item_type, name=name, required=True)
 
 
-def _as_list_item(*, node: InputFormField) -> InputFormField:
+_LIST_ITEM_ADAPTER: TypeAdapter[InputFormItem] = TypeAdapter(InputFormItem)
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/pipeline/input_form.py:510]: Rule refined at its source instead: validating into a top-level union owned by an external contract (here mthds.protocol's InputFormItem) is TypeAdapter's sanctioned use — 5baacb8ee

File: tests/unit/pipelex/pipeline/test_input_form_deriver.py [hunk 2]
```diff
@@ -72,3 +72,57 @@ def test_chain_ending_at_a_base_absent_from_the_crate_is_unknown(self) -> None:
         assert node.kind == FieldKind.UNKNOWN
         assert node.refines == ["dep->other.Score"]
         assert node.description == "Refines a dependency's score"
+
+
+class TestDerivedListItemsAreNameless:
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/pipeline/test_input_form_deriver.py]: Fixed + 5baacb8ee — moved to tests/unit/pipelex/pipeline/test_input_form_list_items.py

**Comment by lchoquel** [comment]
Review round 1 — one comment led to refining the TypeAdapter agent rule at its source (the flagged adapter is the rule's sanctioned boundary case and stays), the test-module split is fixed. Round 2 admits only a regression from these changes or a ship-blocker.
